src/code/components/chromosome.js

```diff
@@ -57,11 +58,14 @@ const ChromosomeView = ({chromosome, chromosomeDescriptor, ChromosomeImageClass=
             }}/>
           );
         } else {
+          let alleleSelected = selectedAlleles && selectedAlleles[chromosome.chromosome] && selectedAlleles[chromosome.chromosome][chromosome.side];
```

Comment:
There's still a small bug here, related to the old implementation. If you make any selection on a chromosome, close and re-open the panel, all the genes on that chromosome will appear selected (even if some weren't before), and all of your selections will have been changed to the correct answer.

This seems to work:
```
          let alleleDef = a;
          let showUnknown = false;
          if (defaultUnknown) {
            const geneName = BioLogica.Genetics.getGeneOfAllele(chromosome.species, a.allele).name;
            const alleleSelected = selectedAlleles && selectedAlleles[chromosome.chromosome] && selectedAlleles[chromosome.chromosome][chromosome.side]
                                    && selectedAlleles[chromosome.chromosome][chromosome.side][geneName];
            if (alleleSelected) {
              alleleDef = {
                allele: alleleSelected,
                editable: true
              };
            } else {
              showUnknown = true;
            }
          }

          return (
            <FVGeneLabelView key={alleleDef.allele} editable={editable && alleleDef.editable} chromosomeDescriptor={chromosomeDescriptor}
                             chromosomeHeight={height} allele={alleleDef.allele} species={chromosome.species} hiddenAlleles={ hiddenAlleles } defaultUnknown={showUnknown}
              onAlleleChange={function (event) {
                                onAlleleChange(alleleDef.allele, event.value);
                             }
            }/>
          );
```